src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -1409,6 +1420,97 @@ def disable_features(cmd, client, resource_group_name, cluster_name, features, k
+    troubleshoot_log_path = os.path.join(os.path.expanduser('~'), '.azure', 'connected8s_troubleshoot.log')
+    utils.setup_logger('connectedk8s_troubleshoot', troubleshoot_log_path)
+    tr_logger = logging.getLogger('connectedk8s_troubleshoot')  # logger for troubleshooting, onto a log file
+
+    # Send cloud information to telemetry
+    send_cloud_telemetry(cmd)
+
+    # setting kubeconfig
+    kube_config = set_kube_config(kube_config)
+
+    # Loading the kubeconfig file in kubernetes client configuration
+    load_kube_config(kube_config, kube_context)
+    configuration = kube_client.Configuration()
+    validate_release_namespace(client, cluster_name, resource_group_name, configuration, kube_config, kube_context)
+    try:
+        latest_connectedk8s_version = utils.get_latest_extension_version()
+        local_connectedk8s_version = utils.get_existing_extension_version()
+        tr_logger.info("Latest available connectedk8s version: {}".format(latest_connectedk8s_version))
+        tr_logger.info("Local connectedk8s version: {}".format(local_connectedk8s_version))
+        from azure.cli.core import __version__ as azure_cli_core_version
+        tr_logger.info("azure-cli-core version installed locally: {}".format(azure_cli_core_version))
+        if latest_connectedk8s_version and local_connectedk8s_version != 'Unknown' and local_connectedk8s_version != 'NotFound':
+            if version.parse(local_connectedk8s_version) < version.parse(latest_connectedk8s_version):
+                print("You have an update pending. You can update the connectedk8s extension to latest v{} using 'az extension update -n connectedk8s'".format(latest_connectedk8s_version))
+
+        try:
+            # Fetch ConnectedCluster
+            connected_cluster = client.get(resource_group_name, cluster_name, raw=True)
+            tr_logger.info("Connected cluster resource: {}".format(connected_cluster.response.content))
+        except Exception as ex:
+            try:
+                if ex.error.error.code == "NotFound" or ex.error.error.code == "ResourceNotFound":
+                    tr_logger.error("Connected cluster resource doesn't exist. " + str(ex))
+                    logger.error("The specified ConnectedCluster resource doesn't exist.")
+            except AttributeError:
+                pass
+            tr_logger.error("Couldn't check the existence of Connected cluster resource. Error: {}".format(str(ex)))
+
+        kapi_instance = kube_client.CoreV1Api(kube_client.ApiClient(configuration))
+        try:
+            pod_list = kapi_instance.list_namespaced_pod(consts.Arc_Namespace)
+            pods_count = 0
+            for pod in pod_list.items:
+                pods_count += 1
+                if pod.status.phase != 'Running':
+                    tr_logger.warning("Pod {} is in {} state. Reason: {}. Container statuses: {}".format(pod.metadata.name, pod.status.phase, pod.status.reason, pod.status.container_statuses))
```

Comment:
Can we have one log file for each container in the pod and package all log files in the final zip? That would ease the debugging process for us. If we keep logging all troubleshooting info i a single log file, it will be very difficult to debug

Reply:
We should also capture container logs and not just container statuses. Error logs in the containers are more useful

Reply:
container logs are part of periscope.

Reply:
@sophsoph321 For visibility.

Reply:
Just a point: If container logs are collected by periscope, then successfully executing periscope becomes very critical. (Most of the times, we ask for arc-agent pod logs from the customer for troubleshooting).

Reply:
That is true. Periscope checks are for the deeper level investigation into the issue. **As per design, Periscope is the integral part of this troubleshoot command.**